### ISSUE TITLE
[RFC] Found no advantages in model fusion methods

### ISSUE THREAD

**Issue by xujli** [TL-System/plato#126]
I use this framework to reproduce some algorithms which are helpful to accelerate convergence in iid and non-iid data distribution setting, such as FedAdp, Fedatt, FedProx, all these algorithms are in your ./examples dir. I ran these algorithms and plot the accuracy on test dataset, and found that the curves have almost no difference. There is no bug or execution confuse when I check the code. So I wonder if you can show me some examples and settings that these algorithms perform a more significant advance on convergence than FedAvg. It would be a great pleasure for me if you give me some brief explanations and advise to reproduce these algorithm.

Best Regards.

**Comment by baochunli** [comment]
This is related to the performance of these algorithms that are designed by the authors of their relevant papers, rather than issues of implementations in Plato. The implementations in examples/ are contributed by various contributors, and serve as references and examples for using Plato as a framework. One possible idea towards showing different results is to use more sophisticated models (rather than MNIST), and to introduce non-iid data distributions (using samplers provided by Plato) with the concentration parameter α defined appropriately in the corresponding Dirichlet distribution.

**Comment by xujli** [comment]
Thanks for your reply, I'll try more settings and models. I just want to get some advise on my results reproduction and I'm not familiar with the rules of Issues. I apologize for my wrong use. If I reproduce some other algorithms correctly, may I commit PR to the Repo?

**Comment by baochunli** [comment]
Of course.

**Comment by xujli** [comment]
Sorry to bother you again, I executed same codes on CPU and GPU respectively(e.g. fedatt in /example). Params is:
(In my understanding, total_silos, institution_class_ids and label_distribution is useless when sampler is not orthogonal).
`algorithm:
  total_silos: 10
  type: fedavg
clients:
  do_test: false
  per_round: 4
  simulation: false
  total_clients: 10
  type: simple
data:
  data_path: ./data
  datasource: MNIST
  institution_class_ids: 0;1;2;3;4;5;6;7;8;9
  label_distribution: noniid
  partition_size: 600
  random_seed: 9
  sampler: noniid
results:
  plot: round&accuracy, round&training_time
  types: accuracy, training_time, round_time
server:
  address: 127.0.0.1
  port: 8001
trainer:
  batch_size: 50
  epochs: 1
  learning_rate: 0.01
  model_name: lenet5
  momentum: 0
  optimizer: SGD
  parallelized: false
  rounds: 50
  target_accuracy: 0.98
  type: basic
  weight_decay: 0.0
`
The results showed that, the model on CPU convergences significantly faster than the model on GPU, this convergence gap occurred in many settings and algorithms, I use APIs to reimplement FedAvg and FedProx, the gap also exists. If it's convenient for you, you can execute the code with setting like above and verify the results.
